openff/interchange/_tests/unit_tests/interop/lammps/export/test_lammps.py

```diff
@@ -62,3 +68,104 @@ def test_to_lammps_single_mols(self, mol, sage_unconstrained, n_mols):
                 "Torsion": 0.02 * unit.kilojoule_per_mole,
             },
         )
+
+    def test_unique_lammps_mol_ids(
+        self, smiles: str, sage_unconstrained: ForceField, sidelen: int = 0
+    ) -> bool:
+        """Test to see if interop.lammps.export._write_atoms() writes unique ids for each distinct Molecule"""
+        temp_lammps_path = (
+            Path.cwd() / "temp.lmp"
+        )  # NOTE: lammps writer doesn't currently support IO-like object passing, so need to intercept output with temporary file
+
+        # BUILD TOPOLOGY
+        ## 0) generate pilot Molecule and conformer
+        pilot_mol = Molecule.from_smiles(smiles)
+        pilot_mol.generate_conformers(n_conformers=1)
+        # pilot_mol.assign_partial_charges(partial_charge_method='gasteiger') # generate dummy charges for testing
+
+        ## 1) compute effective radius as the greatest atomic distance from barycenter (avoids collisions when tiling)
+        conf = pilot_mol.conformers[0]
+        COM = conf.mean(axis=0)
+        conf_centered = conf - COM
+
+        radii = numpy.linalg.norm(conf_centered, axis=1)
+        r_eff = radii.max()
+
+        ## 2) generate 3D integer lattice to tile mols onto
+        xyz_offsets = numpy.column_stack(
+            [  # integral offsets from 0...(sidelen - 1) along 3 axes
+                axis_offsets.ravel()
+                for axis_offsets in numpy.meshgrid(
+                    *[
+                        numpy.arange(sidelen) for _ in range(3)
+                    ],  # the 3 here is for 3-dimensions
+                )
+            ]
+        )
+
+        ## 3) build topology by tiling
+        tiled_top = Topology()
+        for int_offset in xyz_offsets:
+            mol = Molecule.from_smiles(smiles)
+            mol.add_conformer(
+                (conf_centered + 2 * r_eff * int_offset).to("nm")
+            )  # space copied by effective diameter
+            tiled_top.add_molecule(mol)
+
+        ## 3a) set periodic box tightly around extremem positions in filled topology
+        box_dims = tiled_top.get_positions().ptp(axis=0)
+        box_vectors = (
+            numpy.eye(3) * box_dims
+        )  # convert to diagonal matrix to get proper shape
+
+        # EXPORT TO LAMMPS
+        interchange = Interchange.from_smirnoff(
+            sage_unconstrained, tiled_top
+        )  # , charge_from_molecules=[pilot_mol])
+        # UNRELATED QUESTION: why are waters not correctly recognized?
+        # PDB residue name on output is always UNK for Molecule.from_smiles("O"), EVEN when using tip3p.offxml)
```

Comment:
SMIRNOFF force fields don't define residue metadata and our infrastructure doesn't have default residue names for small molecules. As cumbersome as it might seem for simple cases like water, residue metadata needs to be set manually (or programmatically ...) https://github.com/openforcefield/openff-toolkit/issues/1554